### PR TITLE
[Wallet]: Add Payments You've Requested notifications group & screen

### LINKED ISSUE #1744: Grouped Notifications should include outstanding payment requests
### Expected Behavior
When a payment request is sent, a notification should be added to the notification center giving the user the option to either 'remind' or 'cancel' the payment request
[figma link](https://www.figma.com/file/mdNgkyVi2impjf5HRBLq3H/Notifications)
Once the payment request is completed, this notification should be removed from the notification center.

### Current Behavior

payment requests are not displayed.

### PR DESCRIPTION
**Pull request by i1skn**
### Description
- [x] Split Payment Requests into `incoming` and `outgoing`.
- [x] Implement `Cancel` feature on an `outgoing` payment request.
- [x] Implement `Remind` feature on an `outgoing` payment request.
- [x] Fix old tests for `incoming` requests.
- [x] Create new tests for `outgoing` requests.
- [x] Refactor code duplication between `incoming` and `outgoing` requests.
- [x] Add Spanish translations.

### Tested

Ran on Android device

### Related issues

- Fixes #1744

### Backwards compatibility

Yes

### Screenshots
screen1
screen2

### CODE REVIEW

File: packages/mobile/locales/en-US/paymentRequestFlow.json [hunk 1]
```diff
@@ -1,10 +1,13 @@
...
   "requests": "Requests",
   "empty": "You have no Payment Requests",
-  "paymentRequests": "Payment Requests",
+  "incomingPaymentRequests": "Requests Waiting on You",
+  "outgoingPaymentRequests": "Payments You’ve Requested",
   "celoDollarBalance": "{{CeloDollar}} Balance",
   "pay": "Pay",
   "decline": "Decline",
+  "remind": "Remind",
+  "cancel": "Cancel",
```
Comment (jmrossy) [inline packages/mobile/locales/en-US/paymentRequestFlow.json]: Nit: for one word translations that will likely be used elsewhere, please put them in the `global.json` file. You'll find there's already a cancel translation there.

File: packages/mobile/src/account/selectors.test.ts [hunk 2]
```diff
@@ -30,7 +30,38 @@ describe('selectors', () => {
         },
       }
 
-      expect(getPaymentRequests(state)).toEqual([goodRequest])
+      expect(getIncomingPaymentRequests(state)).toEqual([goodRequest])
+    })
+  })
+
+  describe(getOutgoingPaymentRequests, () => {
```
Comment (jmrossy) [inline packages/mobile/src/account/selectors.test.ts:37]: I appreciate the thoroughness and I'm not sure why wrote the existing test above but IMHO selectors and reducers are too simple to be worth testing. Definitely don't remove this now that it's done but for the future I wouldn't bother with this.

File: packages/mobile/src/firebase/saga.ts [hunk 3]
```diff
@@ -153,6 +183,25 @@ export function* watchPaymentRequestStatusUpdates() {
...
+    Logger.debug(TAG, 'Updating payment request', id, `notified: ${notified}`)
+    yield call(() =>
+      firebase
+        .database()
+        .ref(`${REQUEST_DB}/${id}`)
+        .update({ notified })
+    )
+    Logger.debug(TAG, 'Payment request notified updated', id)
+  } catch (error) {
+    Logger.error(TAG, `Error while updating payment request ${id} status`, error)
```
Comment (jmrossy) [inline packages/mobile/src/firebase/saga.ts:199]: we should probably show the user an error here. Please add a new error message and call `yield put showError` here

Comment (jmrossy) [reply packages/mobile/src/firebase/saga.ts:199]: Ditto for above in updatePaymentRequestStatus

File: packages/mobile/src/paymentRequest/OutgoingPaymentRequestListItem.tsx [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+          )}
+          <Text style={[fontStyles.subSmall, styles.comment]}>{this.props.comment}</Text>
+        </View>
+      </BaseNotification>
+    )
+  }
+}
+
+const styles = StyleSheet.create({
+  body: {},
```
Comment (jmrossy) [inline packages/mobile/src/paymentRequest/OutgoingPaymentRequestListItem.tsx]: Nit: remove

File: packages/mobile/src/paymentRequest/OutgoingPaymentRequestListItem.tsx [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+  comment: {
+    paddingTop: variables.contentPadding,
+  },
+  phoneNumber: {
+    color: colors.dark,
+  },
+})
+
+export default componentWithAnalytics(
```
Comment (jmrossy) [inline packages/mobile/src/paymentRequest/OutgoingPaymentRequestListItem.tsx]: Only screens should to be wrapped in `componentWithAnalytics`. Smaller subcomponents don't need it

Comment (jmrossy) [reply packages/mobile/src/paymentRequest/OutgoingPaymentRequestListItem.tsx]: Ditto for the incoming item which incorrectly has that as well

File: packages/mobile/src/paymentRequest/OutgoingPaymentRequestListScreen.tsx [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+    )
+  }
+}
+
+const styles = StyleSheet.create({
+  container: {
+    backgroundColor: colors.background,
+    flex: 1,
+  },
+  separator: {
```
Comment (jmrossy) [inline packages/mobile/src/paymentRequest/OutgoingPaymentRequestListScreen.tsx:98]: There's a HorizontalLine component in react-components/components for this

File: packages/mobile/test/schemas.ts [hunk 7]
```diff
@@ -186,6 +186,30 @@ export const v4Schema = {
     verificationStatus: 0,
     hasSeenVerificationNux: false,
   },
+  account: {
```
Comment (jmrossy) [inline packages/mobile/test/schemas.ts:193]: This is okay but most of these values are duplicating what was already set for `account` above in v0Schema

**Review by jmrossy — COMMENTED** [review]
Excellent work! Just some minor comments

File: packages/mobile/test/schemas.ts (hunk 7 above)
Comment (i1skn) [reply packages/mobile/test/schemas.ts:193]: I wanted to remove `paymentRequests` from the schema, what approach would be better?

File: packages/mobile/src/paymentRequest/OutgoingPaymentRequestListScreen.tsx (hunk 6 above)
Comment (i1skn) [reply packages/mobile/src/paymentRequest/OutgoingPaymentRequestListScreen.tsx:98]: It has unchangeable `width: '100%'`. I could've make it changeable, but on the new design this line will disappear anyway.

File: packages/mobile/locales/en-US/paymentRequestFlow.json [hunk 8]
```diff
@@ -1,10 +1,10 @@
...
-  "paymentRequests": "Payment Requests",
+  "incomingPaymentRequests": "Requests Waiting for You",
+  "outgoingPaymentRequests": "Payments You’ve Requested",
   "celoDollarBalance": "{{CeloDollar}} Balance",
-  "pay": "Pay",
-  "decline": "Decline",
   "requestDeclined": "Request Declined",
-  "requestPaid": "Request Paid"
+  "requestPaid": "Request Paid",
+  "paymentRequestUpdateFailed": "Can not update payment request"
```
Comment (jmrossy) [inline packages/mobile/locales/en-US/paymentRequestFlow.json]: Nit: `Cannot` 1 word or `Error updating payment request`

**Review by jmrossy — APPROVED** [review]

**Pull request merged**